### DISCUSSION TITLE
My session data is lost (laravel/breeze)

### DISCUSSION THREAD

**Discussion by Heveya** [laravel/framework#46735] [Q&A]
I'm using laravel/breeze. In my login function i'm putting user data in my session:
$request->session()->put('ORIGINAL_USER_ID', request()->user()->id);

My .env SESSION_LIFETIME is 1 (1 minute).

**Problem:** In my login form i'm selecting remember me then press submit. Now i have one session file in my storage with ORIGINAL_USER_ID parameter that i've saved when login. After one minute i'm still authorized but now i have no ORIGINAL_USER_ID in my current session. I watch my storage and now i have two session files.

**Questions:**
1. Why the second session file?
2. Why in new session file no ORIGINAL_USER_ID from first session file?

**More information about project.**
In my application users can create children account and then toggle between their accounts in their dashboard panel. Also users can sign in any child account by login form on the main page. When users toggle between their accounts I using Auth::loginUsingId($children_user_id, true) to change authorized user in session. Users don't need type password when they want to toggle between child accounts but I need know what account was authorized first(in login form on the main page). Can you tip me best way to realize this than I do?

**Comment by Angel5a** [comment]
Question 1 looks like an intended behavior. When you use login/logout session key get regenerated.
Question 2 looks like your session expired, so no any data of it should be count as valid, and old values has been dropped.
Check, if both files exists with a proper values right after loginUsingId() request have been processed (no need to wait 1 minute).

What can you do:
If all the "child accounts" belongs to one "original account", you'd be better to store this relationship is a database, so you won't be needed to save anything in session.
If multiple users can share subset of "child accounts", then everything get worse. You need to extend SESSION_LIFETIME to cover your needs.
